opencti-platform/opencti-front/src/private/components/common/stix_core_objects/StixCoreObjectQuickSubscription.tsx

```diff
@@ -0,0 +1,638 @@
+    query StixCoreObjectQuickSubscriptionTriggersLinesPaginationQuery(
+        $search: String
+        $count: Int!
+        $cursor: ID
+        $orderBy: TriggersOrdering
+        $orderMode: OrderingMode
+        $filters: [TriggersFiltering!]
+    ) {
+        myTriggers(
+            search: $search
+            first: $count
+            after: $cursor
+            orderBy: $orderBy
+            orderMode: $orderMode
+            filters: $filters
+        ) @connection(key: "Pagination_myTriggers") {
+            edges {
+                node {
+                    id
+                    name
+                    trigger_type
+                    event_types
+                    description
+                    filters
+                    created
+                    modified
+                    outcomes
+                }
+            }
+        }
+    }
+`;
+
+interface InstanceTriggerEditionFormValues {
+  id: string;
+  name: string;
+  description: string | null;
+  event_types: string[];
+  outcomes: string[];
+}
+
+interface StixCoreObjectQuickSubscriptionProps {
+  instanceId: string,
+  instanceName: string,
+}
+
+const StixCoreObjectQuickSubscription: FunctionComponent<StixCoreObjectQuickSubscriptionProps> = ({ instanceId, instanceName }) => {
+  const classes = useStyles();
+  const { t } = useFormatter();
+  const [open, setOpen] = useState(false);
+  const [displayDelete, setDisplayDelete] = useState<boolean>(false);
+  const [deleting, setDeleting] = useState<boolean>(false);
+
+  const instanceFilters = {
+    instanceId: [{
+      id: instanceId,
+      value: instanceName ?? '',
+    }],
+  };
+  const jsonInstanceFilters = JSON.stringify(instanceFilters);
+  const paginationOptions = {
+    count: 25,
+    filters: [
+      {
+        key: ['filters'],
+        values: [jsonInstanceFilters],
+      },
+      {
+        key: ['trigger_type'],
+        values: ['live'],
+      }],
+  } as StixCoreObjectQuickSubscriptionTriggersLinesPaginationQuery$variables;
+
+  const [commitAddTrigger] = useMutation<TriggerLiveCreationLiveMutation>(
+    triggerLiveAddMutation,
+  );
+  const [commitFieldPatch] = useMutation(triggerMutationFieldPatch);
+  const [commitDeleteTrigger] = useMutation(TriggerPopoverDeletionMutation);
+
+  const existingInstanceTriggers = useLazyLoadQuery<StixCoreObjectQuickSubscriptionTriggersLinesPaginationQuery>(
+    stixCoreObjectQuickSubscriptionTriggersLinesQuery,
+    paginationOptions,
+  );
+  const existingInstanceTriggersEdges = existingInstanceTriggers.myTriggers?.edges ?? undefined;
+  const triggerUpdate = existingInstanceTriggersEdges && existingInstanceTriggersEdges.length > 0;
+
+  const eventTypesOptions: Record<string, string> = {
+    create: t('Creation'),
+    update: t('Modification'),
+    delete: t('Deletion'),
+  };
+  const outcomesOptions: Record<string, string> = {
+    'f4ee7b33-006a-4b0d-b57d-411ad288653d': t('User interface'),
```

Comment:
What are those ids ?

Reply:
Can we have an enum somewhere ? To hidde this ID from component

Reply:
fixed